### PR TITLE
feat: add more controls and cameras

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#409

### CODE REVIEW

File: src/camera/Camera.ts [hunk 1]
```diff
@@ -149,9 +181,9 @@ abstract class Camera {
       return;
     }
 
-    const viewportSize = flicking.viewport.size;
+    const viewport = flicking.viewport;
     const cameraPosition = this._position;
-    const cameraSize = flicking.horizontal ? viewportSize.width : viewportSize.height;
+    const cameraSize = flicking.horizontal ? viewport.width : viewport.height;
```
Comment (mixed) [inline src/camera/Camera.ts:186]: The logic to find the size of the viewport is often seen. How about making it a function?
`flicking.horizontal ? viewport.width : viewport.height;`

File: src/camera/CircularCamera.ts [hunk 2]
```diff
@@ -2,12 +2,186 @@
...
+    return super.lookAt(pos);
+  }
+
+  protected _applyTransform(): void {
+    const el = this._el;
+    const flicking = getFlickingAttached(this._flicking, "Camera");
+
+    el.style[this._transform] = flicking.horizontal
+      ? `translate(${-(this._position - this._alignPos + this._circularOffset)}px)`
+      : `translate(0, ${-(this._position - this._alignPos + this._circularOffset)}px)`;
```
Comment (mixed) [inline src/camera/CircularCamera.ts:166]: This logic is also repeated.

File: src/control/Control.ts [hunk 3]
```diff
@@ -16,6 +18,12 @@ abstract class Control {
   protected _controller: AxesController;
   protected _activePanel: Panel | null;
 
+  public get controller() { return this._controller; }
+  public get activeIndex() { return this._activePanel?.index ?? -1; }
```
Comment (mixed) [inline src/control/Control.ts:22]: Does the meaning of `-1` mean there is no active index? If so, I think you have to some add description.

**Review by mixed — APPROVED** [review]
🥇 🥇 🥇 🥇 🥇

File: src/camera/Camera.ts (hunk 1 above)
Comment (WoodNeck) [reply src/camera/Camera.ts:186]: That's right, I'm considering adding it as a `camera.size` maybe

File: src/control/Control.ts (hunk 3 above)
Comment (WoodNeck) [reply src/control/Control.ts:22]: As a flicking returns `-1` from `getIndex` when there are no panels that exist, that also returns `-1` when there is no active panel.
This is documented on the Flicking's `getIndex`, but not on the Control as there's no description on the Control yet.
I'll add the descriptions while adding the whole documentation of it :)

File: src/renderer/Renderer.ts [hunk 4]
```diff
@@ -182,7 +205,7 @@ abstract class Renderer {
     const align = this._align;
 
     return typeof align === "object"
-      ? (align as { anchor: string | number }).anchor
+      ? (align as { panel: string | number }).panel
```
Comment (daybrush) [inline src/renderer/Renderer.ts:208]: Is there a type other than number, string, or object for align?

Comment (WoodNeck) [reply src/renderer/Renderer.ts:208]: Those three types that you've mentioned are only types that are used for the align option.

**Review by daybrush — APPROVED** [review]
LGTM

**Pull request merged**